src/preferences.jl

```diff
@@ -0,0 +1,19 @@
+@static if VERSION >= v"1.6"
+    using Preferences
+end
+
+@static if VERSION >= v"1.6"
+    const int2float_type = @load_preference("int2float", "Float64")
+else
+    const int2float_type = "Float32"
+end
+
+function set_int2float!(int2float_type::String)
+    if !(int2float_type in ("Float64", "Float32"))
```

Comment:
Since it seems you only intend to support FLoat64 and Float32 in this PR, wouldn't it be simpler to just add a boolean preference for switching between both cases?